extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -25,12 +25,26 @@ export class WorkspaceService implements IWorkspaceService {
 
 	private openedProjects: vscode.Uri[] | undefined = undefined;
 	private excludedProjects: string[] | undefined;
+	private _isProjectProviderAvailable: boolean = false;
 
 	constructor() {
 		Logger.log(`Calling getProjectsInWorkspace() from WorkspaceService constructor`);
+		this.updateIfProjectProviderAvailable();
 		this.getProjectsInWorkspace(undefined, true).catch(err => Logger.error(`Error initializing projects in workspace ${err}`));
 	}
 
+	get isProjectProviderAvailable(): boolean {
+		return this._isProjectProviderAvailable;
+	}
+
+	public updateIfProjectProviderAvailable(): void {
+		Logger.log(`Checking ${vscode.extensions.all.length} extensions to see if there is a project provider is available`);
+		const startTime = new Date().getTime();
+
+		this._isProjectProviderAvailable = vscode.extensions.all.find(e => e.packageJSON.contributes?.projects?.length > 0) !== undefined;
```

Comment:
Getting my helper functions mixed up, some is the one you'd use for this kind of scenario.

```suggestion
		this._isProjectProviderAvailable = vscode.extensions.all.some(e => e.packageJSON.contributes?.projects?.length > 0);
```